src/libexpr/primops/fetchTree.cc

```diff
@@ -182,45 +182,208 @@ static RegisterPrimOp primop_fetchTree({
+           > }
+           > ```
+
+      - `"git"`
+
+        Fetch a Git tree and copy it to the Nix store.
+        This has the same underyling implementation as [`builtins.fetchGit`](@docroot@/language/builtins.md#builtins-fetchGit)
+
+        - `url` (String, required)
+
+          The URL formats supported are the same as for Git itself.
+
+          > **Example**
+          >
+          > ```nix
+          > fetchTree {
+          >   type = "git";
+          >   url = "git@github.com:NixOS/nixpkgs.git";
+          > }
+          > ```
+
+        - `ref` (String, optional)
+
+          A [Git reference](https://git-scm.com/book/en/v2/Git-Internals-Git-References), such as a branch or tag name.
+
+        - `rev` (String, optional)
+
+          A Git revision, typically a commit hash.
+
+        - `shallow` (Bool, optional)
+
+          Make a shallow clone when fetching the Git tree.
+
+          Default: `false`
+
+        - `submodules` (Bool, optional)
+
+          Also fetch submodules if available.
+
+          Default: `false`
+
+        - `allRefs` (Bool, optional)
+
+          If set to `true`, always fetch the entire repository, even if the latest commit is still in the cache.
+          Otherwise, only the latest commit is fetched if it is not already cached.
+
+          Default: `false`
+
+        - `lastModified` (Integer, optional)
+
+          Unix timestamp of the fetched commit.
+
+          If set, pass through the value to the output attribute set.
+          Otherwise, generated from the fetched Git tree.
+
+        - `revCount` (Integer, optional)
+
+          Number of revisions in the history of the Git repository before the fetched commit.
+
+          If set, pass through the value to the output attribute set.
+          Otherwise, generated from the fetched Git tree.
+
+      > **Note**
+      >
+      > The following input types are **experimental** and may be subject to change:
+      >
+      > - `"path"`
+      > - `"github"`
+      >   - `owner`
+      >   - `repo`
+      > - `"gitlab"`
+      > - `"sourcehut"`
+      > - `"mercurial"`
+      >
+      > Enable the [`flakes` experimental feature](@docroot@/contributing/experimental-features.md#xp-feature-flakes) to use them.
+      >
+      > With flakes enabled, the input should be *locked*.
+      > That is, it should contain a commit hash or content hash, unless impure evaluation (`--impure`) is enabled.
+
+      > **Note**
+      >
+      > *input* can also be a [URL-like reference](@docroot@/command-ref/new-cli/nix3-flake.md#flake-references).
+      > The URL-like syntax requires the [`flakes` experimental feature](@docroot@/contributing/experimental-features.md#xp-feature-flakes) to be enabled.
+
+      > **Example**
+      >
+      > Fetch a GitHub repository using the attribute set representation:
+      >
+      > ```nix
+      > builtins.fetchTree {
+      >   type = "github";
```

Comment:
Better not use something that's experimental as an example.

Reply:
There's a good chance we'll stabilize these together, considering that possibly a majority of usages of fetchTree are with this type.